iter5

shestooy opened the pull request.

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
initial commit

Commit:
-add new test for handlers_test

Commit:
-fix error in main package of server

Commit:
- fix bug in handlers test

Commit:
- fix bug in main agent

Commit:
- fix name of function

Commit:
- fix bug in GetMetricID

Commit:
- add response close body in handlers_test.go

Commit:
- delete README.md

shestooy committed:
initial commit

shestooy committed:
initial commit

shestooy committed:
default server flag value changed to "localhost:8080"

shestooy committed:
fix usage of flags

shestooy committed:
add prefix http in post order

shestooy committed:
add new test
change logic in main agent

shestooy committed:
change error answer by server

shestooy committed:
added the ability to work with env variables (iter 5)

shestooy commented:
Привет, прошу прощения название некоторых коммитов, буду исправляться)

Bazys commented on cmd/agent/main.go:
```diff
@@ -1,3 +1,121 @@
+
+var pollCount int64
+
+func getAllMetrics() []Metric {
```
давай выносить всю логику из main.
Это точка входа - здесь минимум кода - только для запуска.

Bazys commented on cmd/server/main.go:
```diff
@@ -1,3 +1,40 @@
+	return http.ListenAndServe(serverEndPoint, GetRouter())
+}
+
+func GetRouter() chi.Router {
```
тоже перенсти в internal

Bazys commented on internal/handlers/handlers.go:
```diff
@@ -0,0 +1,120 @@
+	}
+}
+
+func GetAllMetrics(res http.ResponseWriter, req *http.Request) {
```
Параметр функции, названный `req`, не используется в теле функции. В Go, если определили параметр в функции, но не использовали его внутри, это считается неоптимальным стилем, поскольку параметр занимает место в сигнатуре функции и может сбивать с толку при чтении кода.
В данном случае, есть два возможных решения:
1. **Удаление параметра:** Если параметр `req` действительно не нужен для работы функции, его можно просто удалить из сигнатуры функции. Это делает код чище и проще для понимания.
2. **Переименование в `_`:** Если параметр не используется, но его наличие обязательно по какой-то причине (например, для соответствия интерфейсу или для использования в будущем), его можно переименовать в `_`. В Go, символ `_` используется для игнорирования переменной. Переменные, именованные как `_`, не обязаны быть использованы, т.е. не будет копирования неиспользуемого значения внутрь функции. Это позволяет сохранить сигнатуру функции неизменной, но явно указывает на то, что данный параметр в настоящее время не используется и избежать лишнего использования памяти.

Bazys reviewed (changes requested):
main - точка входа - тут минимум кода

Commit:
the logic has been moved from cmd to internal.
packages added:
-agent
-flags
--http server
-metrics

Tests have been added to the packages.

Commit:
added storage initialization in test agent package (previously missed)

Bazys commented on cmd/server/main.go:
```diff
@@ -1,3 +1,15 @@
+)
+
+func main() {
+	flags.ParseServerFlags()
```
```suggestion
	flags.ParseServer()
```
flags - уже есть, зачем повторять?

shestooy replied:
тут немного не понял, что исправить

Bazys commented on internal/agent/agent.go:
```diff
@@ -0,0 +1,63 @@
+	reportTicker := time.NewTicker(time.Duration(f.ReportInterval) * time.Second)
+	defer reportTicker.Stop()
+
+	metrics := make([]m.Metric, 0)
```
не очень рационально с точки зрения работы с памятью, но во втрой половине будет pptof - там посмотришь

Bazys commented on internal/flags/flags.go:
```diff
@@ -0,0 +1,41 @@
+		AgentEndPoint = envAgentEndPoint
+	}
+	if envReportInterval := os.Getenv("REPORT_INTERVAL"); envReportInterval != "" {
+		ReportInterval, _ = strconv.ParseInt(envReportInterval, 10, 64)
```
почему ошибки не обрабатываем?

Bazys reviewed (approved):
щас заворачивать не буду, но к следующему разу исправить

Commit:
added handlers for errors
